vault/cors.go

```diff
@@ -0,0 +1,112 @@
+package vault
+
+import (
+	"errors"
+	"net/http"
+	"strings"
+)
+
+var errCORSNotConfigured = errors.New("CORS is not configured")
+
+var responseHeaders = map[string]string{
+	"Access-Control-Allow-Headers":     "origin,content-type,cache-control,accept,options,authorization,x-requested-with,x-vault-token",
+	"Access-Control-Max-Age":           "1800",
+	"Access-Control-Allow-Credentials": "true",
+	"Vary": "Origin",
+}
+
+var allowedMethods = []string{
+	http.MethodDelete,
+	http.MethodGet,
+	http.MethodOptions,
+	http.MethodPost,
+	http.MethodPut,
```

Comment:
Yes, but that's not an official method, so it won't be in the `http` package.

I have a similar issue here though as with the pre-set list of response headers. Why are we artificially limiting? Why not set `*` and let the admin change later if needed?